django/contrib/postgres/indexes.py

```diff
@@ -1,10 +1,13 @@
+from django.db import connection
 from django.db.models import Index
 
 __all__ = ['BrinIndex', 'GinIndex']
 
 
 class BrinIndex(Index):
     suffix = 'brin'
+    # The connection for this index should always be PostgreSQL
+    max_name_length = connection.ops.max_name_length()
```

Comment:
Maybe we should access `connection` in a `cached_property()` to prevent model level query? e.g. what if a backend was executing a query to determine the `max_name_length()` because it was based on a compilation flag?